### PR TITLE
`General`: Add links to the Artemis documentation for more context

### PR DESCRIPTION
**Pull request by sebastiannikles**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

During a user study regarding usability of Artemis, it became apparent that especially new users don't understand the meaning of certain terms and what/how they can interact with them.
Since there is a documentation for Artemis, which provides a lot more context about the different parts of Artemis, which could help those users understand the platform better. Right now, accessing the documentation is not really obvious when using Artemis.

### Description

In order to make accessing the documentation easier, I added a help/information button to relevant elements (such as courses, exercises, exams, lectures, ...), which open the corresponding page in the documentation when being clicked.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to different pages such as courses, lectures, units, learning goals, exercises
3. Make sure the information button is visible 
4. Click the button
5. Make sure the corresponding page of the documentation opens

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

The information button

Bildschirmfoto 2022-11-02 um 09 49 34

The opened documentation
Bildschirmfoto 2022-11-02 um 09 49 41

### CODE REVIEW

**Review by LiaLism — DISMISSED** [review]
The links worked for courses, exercises, lectures, etc. Tested on ts1

**Review by timo07k — APPROVED** [review]
Links work everywhere, tested on TS1

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
I'd expect them in a few more places, especially the student view.
- Each exercise detail page should contain a link to that specific exercise type (maybe in the header?)
- As an instructor I see the information button in the exercise overview in the course management but not in the exam overview
- The information for the communication is a bit hidden
- I also don't see an information button for a tutor trying to assess exercises. This is also something we have to explain the new tutors each year

**Review by tunargs — DISMISSED** [review]
Tested on ts1. Links are working

**Review by PhilipWerz — DISMISSED** [review]
All links **worked fine** for exercise creation (programming, Quiz etc.), Exams, Lectures and Learning Goals. Tested on ts1.

Remark:

1. No help button when creating a lecture (only at the lectures overview page)
2. same for learning goals
3. No help button in "Create Tutorial Group"

**Comment by sebastiannikles** [comment]
I added the button to a lot more places as suggested by you.

I didn't really add it for the student views because I feel like the docs contain primarily information about how to set up things in Artemis and hence, they are important for tutors/instructors who create content.

**Review by manuelmanso — CHANGES_REQUESTED** [review]
Two requests/suggestions:
- Add a tooltip for the documentation icon, otherwise it's not really clear where the user will be taken and why they should click it
- It should be clear for developers that the routes in **documentation-button.component** need to match the documentation routes so we dont break anything when changing documentation. Maybe a comment there is enough?

**Review by evgeny1984 — CHANGES_REQUESTED** [review]
Improvement suggestions:
- Add the link to the documentation for the statistics page as well

File: src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Component, Input } from '@angular/core';
+import { faCircleInfo } from '@fortawesome/free-solid-svg-icons';
+
+// The routes here are used to build the link to the documentation.
+// Therefore, it's important that they exactly match the url to the subpage of the documentation.
+// Additionally, the case names must match the keys in documentationLinks.json for the tooltip.
+export enum DocumentationType {
+    Course = <any>'courses/customizable/',
```
Comment (manuelmanso) [inline src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts:8]: Are these <any> necessary?

File: src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { Component, Input } from '@angular/core';
```
Comment (manuelmanso) [inline src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts:1]: Please add a simple test for this component

File: src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    Assessment = <any>'exercises/assessment/',
+    Statistics = <any>'learning-analytics/',
+}
+
+@Component({
+    selector: 'jhi-documentation-button',
+    styleUrls: ['./documentation-button.component.scss'],
+    template: `
+        <button type="button" class="text-primary documentation-button" (click)="openDocumentation()">
+            <fa-icon [icon]="faCircleInfo" ngbTooltip="{{ getTooltipForType() | artemisTranslate }}"></fa-icon>
```
Comment (manuelmanso) [inline src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts:32]: Can you use our `jhi-help-icon.component` here?

**Review by manuelmanso — CHANGES_REQUESTED** [review]
Tested in ts1, still works and tooltip makes it a lot more understandable. Please check the feedback below

File: src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts (hunk 1 above)
Comment (sebastiannikles) [reply src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts:8]: Yes, without them we couldn't reverse map the type to its case name. We need the name for the translation string

File: src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts (hunk 3 above)
Comment (sebastiannikles) [reply src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts:32]: It's not the same design so I would not use it. Also in both cases it's 1 line so I think it's fine with using my own icon

**Review by manuelmanso — DISMISSED** [review]
Thanks for addressing my feedback, code looks good!

**Review by evgeny1984 — APPROVED** [review]
The code looks good to me. The links are there. Thank you for integrating the requested changes.

**Review by evgeny1984 — APPROVED** [review]
Reapproved. Works as expected, tested on ts1.

**Comment by krusche** [comment]
@sebastiannikles today we just changed the URL of the documentation to https://docs.artemis.cit.tum.de
Can you please update this in your PR in case you use it

File: src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    Course = <any>'courses/customizable/',
+    Lecture = <any>'lectures/',
+    Exercise = <any>'exercises/',
+    Quiz = <any>'exercises/quiz/',
+    Model = <any>'exercises/modeling/',
+    Programming = <any>'exercises/programming/',
+    Text = <any>'exercises/textual/',
+    FileUpload = <any>'exercises/file-upload/',
+    Notifications = <any>'notifications/',
+    LearningGoals = <any>'lectures/#learning-goals',
```
Comment (rriyaldhi) [inline src/main/webapp/app/shared/components/documentation-button/documentation-button.component.ts:17]: As far as I'm aware, there is no #learning-goals section on the [lectures page](https://docs.artemis.cit.tum.de/user/lectures/#).

**Review by danielbode — APPROVED** [review]
All the links work. Nice addition to Artemis!

**Review by pellumbbaboci — APPROVED** [review]
Tested during testing session all the links work as expected.

**Pull request merged**